### PR TITLE
Give trials admin panel

### PR DESCRIPTION
**Pull request by tspenov**
image
image

### CODE REVIEW

File: lib/sanbase_web/admin/billing/promo_trial.ex [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        input(promo_trial, :trial_days)
+      end
+
+      inputs "Plans" do
+        inputs(:plans,
+          as: :check_boxes,
+          collection:
+            PromoTrial.promo_trial_plans()
+            |> Enum.map(fn id ->
+              Repo.get(Plan, id) |> Repo.preload(:product)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/admin/billing/promo_trial.ex]: Let's avoid using `Repo` outside the model module itself - `Plan.by_id/1` can be used here

File: lib/sanbase_web/admin/billing/promo_trial.ex [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      attributes_table(all: true)
+    end
+
+    controller do
+      after_filter(:create_promo_trials, only: [:create])
+    end
+  end
+
+  def create_promo_trials(conn, params, resource, :create) do
+    # promo_trial: %{plans: ["13", "43"], trial_days: "23", user_id: 30},
```
Comment (IvanIvanoff) [inline lib/sanbase_web/admin/billing/promo_trial.ex]: leftover

File: lib/sanbase/billing/subscription/promo_trial.ex [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      :trial_days,
+      :plans
+    ])
+    |> stringify_plans(attrs)
+  end
+
+  def promo_trial_plans, do: @promo_trial_plans
+
+  def create_promo_trial(%{plans: plans, trial_days: trial_days, user_id: user_id}) do
+    user = Repo.get(User, user_id)
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/subscription/promo_trial.ex]: User.by_id

File: lib/sanbase/billing/subscription/promo_trial.ex [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  require Logger
+
+  alias Sanbase.StripeApi
+  alias Sanbase.Auth.User
+  alias Sanbase.Billing.{Subscription, Plan}
+  alias Sanbase.Repo
+
+  @plan_id_name_map %{
+    "3" => "Neuro by Santiment / PRO",
+    "13" => "Sanabse by Santiment / PRO",
```
Comment (spiderjako) [inline lib/sanbase/billing/subscription/promo_trial.ex]: Sanabse -> Sanbase

File: lib/sanbase_web/admin/billing/promo_trial.ex [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  register_resource PromoTrial do
+    action_items(only: [:new, :show])
+
+    form promo_trial do
+      inputs do
+        input(promo_trial, :user,
+          collection: Repo.all(User),
+          fields: [:email, :username]
+        )
```
Comment (IvanIvanoff) [inline lib/sanbase_web/admin/billing/promo_trial.ex]: Lately (mainly for `Project`) I prefer to have a button on the project page which leads to another page where the project id is already chosen. This way you can avoid long lists and searching users from a dropdown
image
image

File: lib/sanbase/billing/subscription/promo_trial.ex [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    end
+  end
+
+  defp promotional_subsciption_data(user, plan, trial_days) do
+    %{
+      customer: user.stripe_customer_id,
+      items: [%{plan: plan.stripe_id}],
+      trial_end: Timex.shift(Timex.now(), days: trial_days) |> DateTime.to_unix()
+    }
+    |> IO.inspect()
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/subscription/promo_trial.ex]: leftover

File: lib/sanbase_web/admin/billing/promo_trial.ex [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    controller do
+      after_filter(:create_promo_trials, only: [:create])
+    end
+  end
+
+  def create_promo_trials(conn, params, resource, :create) do
+    # promo_trial: %{plans: ["13", "43"], trial_days: "23", user_id: 30},
+    Sanbase.Billing.Subscription.PromoTrial.create_promo_trial(params.promo_trial)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/admin/billing/promo_trial.ex]: we can simply do `{:ok, _} = ... ` in order for the person trying to do it via the ex_admin to see there's a failure. It's ok to just crash here I guess?

File: lib/sanbase_web/admin/billing/promo_trial.ex (hunk 5 above)
Comment (tspenov) [reply lib/sanbase_web/admin/billing/promo_trial.ex]: yeah, thats better - though I couldn't set it up for some reason.
I use this add on: https://chrome.google.com/webstore/detail/dropdown-search/kgjaddbcmpmacilgaekjfelhdebfjgib for searching in select list

**Pull request merged**